katcp/resource_client.py

```diff
@@ -55,6 +55,23 @@ def _transform(f):
     future.add_done_callback(_transform)
     return new_future
 
+def until_all(futures):
+    """Run multiple operations in parallel but only raise a single TimeoutError.
+
```

Comment:
I'm fine with everything, except until_all seems a bit uneccesary to me. until_any was only introduced since tornado did not have  a similar function. Arguably it has been superceded by more recent tornado functionality. 

Calling it until_all is confusing IMHO, since that is also what I would call the default tornado yield [f1, f2, f...] behaviour, with slight differences in multiple exception handling. 

I would recommend just using  tornado.gen.multi directly. It is well documented by tornado, etc.